vnext/Microsoft.ReactNative/Modules/DeviceInfoModule.cpp

```diff
@@ -102,19 +131,24 @@ void DeviceInfoHolder::updateDeviceInfo() noexcept {
 
     m_windowWidth = window.Bounds().Width;
     m_windowHeight = window.Bounds().Height;
-  } else {
-    /// TODO: WinUI 3 Island - mock for now
-    m_windowWidth = 600;
-    m_windowHeight = 800;
+  } else if (auto hwnd = (HWND)XamlUIService::GetIslandWindow(m_context->Properties())) {
+    RECT rect{};
+    if (CALL_INDIRECT(L"user32.dll", GetWindowRect, hwnd, &rect)) {
+      m_windowWidth = (float)(rect.right - rect.left);
+      m_windowHeight = (float)(rect.bottom - rect.top);
+    }
   }
+
   winrt::Windows::UI::ViewManagement::UISettings uiSettings;
   m_textScaleFactor = uiSettings.TextScaleFactor();
-  auto const displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
-  m_scale = static_cast<float>(displayInfo.ResolutionScale()) / 100;
-  m_dpi = displayInfo.LogicalDpi();
-  m_screenWidth = displayInfo.ScreenWidthInRawPixels();
-  m_screenHeight = displayInfo.ScreenHeightInRawPixels();
-  notifyChanged();
+  if (!react::uwp::IsWinUI3Island()) {
+    auto const displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
+    m_scale = static_cast<float>(displayInfo.ResolutionScale()) / 100;
+    m_dpi = displayInfo.LogicalDpi();
+    m_screenWidth = displayInfo.ScreenWidthInRawPixels();
+    m_screenHeight = displayInfo.ScreenHeightInRawPixels();
```

Comment:
Why do we not set these fields for WinUI 3? Having them to be zeros by default does not seem to be right.